client/acquire_token.go

```diff
@@ -706,3 +735,108 @@ func AcquireToken(destination *url.URL, dirResp server_structs.DirectorResponse,
 
 	return token.AccessToken, nil
 }
+
+// Given a URL and a known public key, determine whether the public key
+// is valid for the issuer URL.
+//
+// If valid, returns the corresponding keyId and sets found to true.
+func findKeyId(url string, ecPubKey *ecdsa.PublicKey) (keyid string, found bool) {
```

Comment:
This seems generically useful enough that it might be better suited as an exported function in one of our utils or token packages. Moving it to utils would try a bit of restructuring because we generally can't import config there due to cyclic dependencies, so I won't push back if you think the function should remain unexported in `client`.

Reply:
Yes -- I'd like to leave the restructuring until we have a second use case.

Since this is a non-public interface, there won't be any compatibility concerns if we wait until later.